docs/use_cases/recommender_systems.md

```diff
@@ -1,15 +1,84 @@
+
+
+### Recommender Systems as Graphs
+
+Before we go into details on how this is achieved (at you’d be surprised how easily!), let’s take a step back and try to observe the typical matrix factorization approach through the lens of graph algorithms. The user-item interaction matrix can be observed as an adjacency matrix of a bipartite graph (‘bi’, because there are two groups of nodes, users and items, and link can exist only between nodes belonging to different groups). The rating recovered (predicted) by the MF approach can be seen as the predicted weight of the edge existing between that particular user-item pair. So, in the end, the recommendations for a user are actually the most likely links this user is going to establish in the future (with the information on the weight of these links).
+
+(We deliberately took a detour and observed the problem via graphs. Note that this approach is known as ‘joint matrix factorization’ in the literature).
+
+Once the problem is understood as inherently a graph problem, it is easy to see how additional information can be added. What we want to achieve is to somehow let the algorithm know about the new links, which would help group similar items or users together. In other words, we want to ‘inject’ new nodes, which would link the nodes belonging to the same group. How could this be achieved? For the sake of simplicity, let’s assume that only genre information is available on the items. In this case, for each genre, we add a dummy user whose interactions are items that belong to this genre. In such a way, given a graph of user and item nodes, the new dummy user allows for coupling of same-genre items, which in turns helps the model.
+
+
+#### Illustration
+
+Have a look at the illustration above. There are three users, U1, U2 and U3, and four items, I1, I2, I3, I4. The user U1 has interacted with items I1 and I3. Now, there is a dummy user, UD, who links the items that have the same color (I1, I3, I4). This increases the chances of item I4 to be recommended to user U1.
```

Comment:
somehow the illustration and it's description need to be tied better with the previous paragraph that formulates the graph interpretation of the problem - it feels disconnected right now

also, I'd suggest using clearer labels - for example "New User" instead of "UD"